File: medium/Linked List/Add Two Numbers.py
class Solution:
    def addTwoNumbers(self, l1, l2):
        ans = res = ListNode(0)
        n1 = n2 = ""
        while l1 or l2:
            if l1:
                n1 = str(l1.val) + n1
                l1 = l1.next
            if l2:
                n2 = str(l2.val) + n2
                l2 = l2.next
        r = int(n1)+int(n2)
        if r==0:
            return res
        while r!=0:
            x = r%10
            r /= 10
            ans.next = ListNode(x)
            ans = ans.next
        return res.next

        
        

File: medium/linked_list/Add Two Numbers.cpp
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode* ans = new ListNode(0);
        ListNode* res = ans;int total =0,carry =0;
        while(l1  || l2 || carry !=0){
            total = carry;
            if(l1){
                total += l1->val;
                l1 = l1->next;
            }if(l2){
                total += l2->val;
                l2 = l2->next;
            }int num = total %10;
            carry = total/10;
            ans->next = new ListNode(num);
            ans = ans->next;
        }return res->next;
    }
};
